azext_iot/commands.py

```diff
@@ -60,13 +62,32 @@ def load_command_table(self, _):
         g.generic_update_command('update', getter_name='iot_device_configuration_show',
                                  setter_name='iot_device_configuration_update', command_type=iotext_custom)
 
-    with self.command_group('iot device') as g:
+    with self.command_group('iot device', client_factory=iot_hub_service_factory) as g:
         g.custom_command('send-d2c-message', 'iot_device_send_message')
         g.custom_command('simulate', 'iot_simulate_device')
-        g.custom_command('upload-file', 'iot_device_upload_file')
 
-    with self.command_group('iot device c2d-message') as g:
+    with self.command_group('iot device c2d-message', client_factory=iot_hub_service_factory) as g:
         g.custom_command('complete', 'iot_c2d_message_complete')
         g.custom_command('abandon', 'iot_c2d_message_abandon')
         g.custom_command('reject', 'iot_c2d_message_reject')
         g.custom_command('receive', 'iot_c2d_message_receive')
+        g.custom_command('receive-mqtt', 'iot_c2d_message_receive_mqtt')
+
+    with self.command_group('iot dps enrollment' , client_factory=iot_service_provisioning_factory) as g:
+        g.custom_command('create', 'iot_dps_device_enrollment_create')
+        g.custom_command('list', 'iot_dps_device_enrollment_list')
+        g.custom_command('show', 'iot_dps_device_enrollment_get')
+        g.custom_command('update', 'iot_dps_device_enrollment_update')
```

Comment:
Were you going to use generic_update_command style for consistency in entity updates?

Reply:
not planning to. See the comments below.